### PR TITLE
Post return error

### PR DESCRIPTION
**Pull request by ethen001**
### Overview of PR

#72 
* getPostByID returns 404 code error when posts are not found with given post information
### Technical Changes
* modified ViewPostByPostId method in forum services to returns null if post info not found
* modified getPostByID to return error when VewPostByPostID returns null
* modified fomumServices and getPostByID test suite to consider these new cases

### CODE REVIEW

File: src/app/user/getPostByID.js [hunk 1]
```diff
@@ -6,6 +6,9 @@ export default async function getPostByID(req, res) {
     const { query, userID: posterID } = req;
     const { postID } = query;
     const post = await viewPostsByPostID({ postID, posterID });
+    if (post === null) {
+      return res.status(httpStatus['404']).jsend.error('POST NOT FOUND');
+    }
```
Comment (songrenzhao) [inline src/app/user/getPostByID.js]: Should never include any logic in controllers. Please handle this logic in the service file

File: src/app/user/__tests__/getPostByID.js [hunk 2]
```diff
@@ -28,6 +28,13 @@ describe('getPostByID test suite', () => {
     expect(res.jsend.success).toBeCalled();
   });
 
+  it('Should catch post not found error', async () => {
```
Comment (songrenzhao) [inline src/app/user/__tests__/getPostByID.js]: Please remove the test case

File: src/services/forumService.js [hunk 3]
```diff
@@ -47,6 +47,9 @@ export async function viewPostsByPostID(postInfo) {
   try {
     const { postID, posterID } = postInfo;
     const post = await findPostsByPostID({ postID });
+    if (!post) {
+      return null;
```
Comment (songrenzhao) [inline src/services/forumService.js]: Can throw an error here. Consider
```js
if (!post) {
    throw new Error('Post Not Found');
}
```
The catch will handle the error and will display a 500 status code with error message of `Post Not Found`

File: src/services/__tests__/forumService.js [hunk 4]
```diff
@@ -69,6 +69,13 @@ describe('Forum service test suite', () => {
       expect(response).toBeDefined();
     });
 
+    it('Should not show post information with non-existing post information', async () => {
+      findPostsByPostID.mockReturnValue(null);
+      const response = await viewPostsByPostID({ posterID: 12345, postID: 123 });
+      expect(findPostsByPostID).toHaveBeenCalledTimes(1);
+      expect(response).toBe(null);
+    });
+
```
Comment (songrenzhao) [inline src/services/__tests__/forumService.js]: Please modify this test case into
```js
it('Should not show post information with non-existing post information', async () => {
  try {
    findPostsByPostID.mockReturnValue(null);
    await viewPostsByPostID({ posterID: 12345, postID: 123 });
  } catch (err) {
    expect(findPostsByPostID).toHaveBeenCalledTimes(1);
    expect(err.message).toBe('Post Not Found');
  }
});
```

Comment (ethen001) [reply src/services/__tests__/forumService.js]: Done

File: src/app/user/__tests__/getPostByID.js (hunk 2 above)
Comment (ethen001) [reply src/app/user/__tests__/getPostByID.js]: Done

File: src/app/user/getPostByID.js (hunk 1 above)
Comment (ethen001) [reply src/app/user/getPostByID.js]: I implemented your suggestions below which means that 'Post Not Found' is returns with 500 code as all error. Please look at my code

**Pull request merged**